modules/programs/fff.nix

```diff
@@ -0,0 +1,582 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.fff;
+
+in {
+  meta.maintainers = [ maintainers.gigahawk ];
+
+  options.programs.fff = {
+    enable = mkEnableOption "A simple file manager written in bash";
+
+    cdOnExit = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Change directory to current folder on exit
+        '';
+      };
+
+      name = mkOption {
+        type = types.str;
+        default = "f";
+        description = ''
+          Name of function used to call fff.
+        '';
+      };
+
+      commands = mkOption {
+        type = types.str;
+        default = ''
+          # cd to exited folder for bash and zsh
+          fff "$@"
+          cd "$(cat "''${XDG_CACHE_HOME:=''${HOME}/.cache}/fff/.fff_d")" 
+        '';
+        example = "";
+        description = ''
+          Bash/Zsh commands used to start fff with cdOnExit enabled.
+        '';
+      };
+
+      fishCommands = mkOption {
+        type = types.str;
+        default = ''
+          fff $argv
+          cd (cat ${cfg.cdOnExit.helperFile})
+        '';
+        description = ''
+          Fish commands used to start fff with cdOnExit enabled.
+        '';
+      };
+
+      helperFile = mkOption {
+        type = types.str;
+        default = "${config.home.homeDirectory}/.cache/fff/.fff_d";
+        description = ''
+          Location of CD on exit helper file
+        '';
+      };
+    };
+
+    config = {
```

Comment:
I think you can remove most of the options here (or possibly choose a few of the most important options). Then introduce a `settings` option of type `attrsOf str` as per [RFC 42](https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md) that users can use.

Reply:
I'm not quite sure how exactly to apply that RFC to this situation, configs are meant to be exported as shell variables, it wouldn't make a lot of sense to have these settings exported to a separate file and then somehow sourced as part of shell start up.

The motivation for having the options explicitly defined is to prevent arbitrary settings from being set here since they will end up as global shell variables.